documentation/AuthoringSolutionsTutorial.md

```diff
@@ -0,0 +1,342 @@
+
+The GPII has several matchmaker plugins, and it's likely a component that will develop more intelligence over time. For now, we'll just list the fact that we're JAWS, and available to the system as such.
+
+```json
+"capabilities": [
+    "applications.com\\.freedomscientific\\.jaws.id"
+]
+```
+
+As you can see, it's just our solution ID between `applications.` and `.id` with an extra level of escaping within the dots of our ID.
+
+TODO Remember why it looks like this, and what the postfix `.id` is for.
+
+#### Transformations
+
+The next section is `capabilitiesTransformations`. We will go over it in some detail here, and then provide links to further reading and documentation. It's a very important part of adapting your solution to the broad needs of users, but can be complicated at first.
+
+We will look at two users, one who explicity lists JAWS in their needs and preferences, and another one who's usage of JAWS is inferred from common terms they have specified.
+
+First, let's assume we have a user with the following preferences:
+
+```json
+"preferences": {
+    "http://registry.gpii.net/applications/com.freedomscientific.jaws": {
+        "cloud4allVoiceProfile-GlobalContext.Speed": 115
+    }
+}
+```
+
+In this case they are specifying that JAWS is a solution they want to use whenever it's available. Inside of that preference they can list specific preferences that should be getting set. Here we have a single property `cloud4allVoiceProfile-GlobalContext.Speed`. In the INI file this will translate into a property `Speed` inside of the section `cloud4allVoiceProfile-GlobalContext`.
+
+Because it is a direct reference to one of the properties, the exact value 115 is used as is.
+
+TODO: Explain why the full registry URL is required here instead of just the application ID.
+
+At this point we'll mention as an aside that this is a minor workaround (and why this is solution example isn't production ready), in that ahead of time we're creating a JAWS profile called `cloud4allVoiceProfile`. This is a case you may run into sometimes, are issues that must be dealt with the first time an application is launched.
+
+Let's take a look at our transforms now, and a profile that uses them. But first, what are we transforming? We are transforming the values of a users needs and preferences that are specified in common terms, to the *exact* values required by the application.  As an example of a measurement we're all familier with, imagine that there was a common term whose measurement was of tempature, and the units for the common term were in celcius.  Now, what if there was an application that required it's analogous preference in farenheit? We'd have to transform it from Celcius to Farenheit, and that's the exact idea behind what's happening here.
```

Comment:
celsius -> Celsius
farenheit & Farenheit-> Fahrenheit
familier -> familiar
it's analogous preference -> its ...